src/directive/validators.ts

```diff
@@ -0,0 +1,24 @@
+import Joi from "joi";
+import type { CreateDirectiveRequest } from "./model";
+
+export const validateCreateDirectiveRequest = (
+  data: unknown
+): data is CreateDirectiveRequest => {
+  const validation = Joi.object()
+    .keys({
+      archiveId: Joi.number().integer().min(1).required(),
+      stewardAccountId: Joi.number().integer().min(1).required(),
```

Comment:
This answers my test question - this requires the steward's id always, even though we only need it for "transfer" type directives, which of course are the only kind we have right now. I think Joi does handle "if this value then this other value" though. What do you think about using something like that here, so the steward is required if the type is "transfer"? Too complicated for the moment?

Reply:
Good idea, will update